sql/src/test/java/io/crate/execution/dml/upsert/ShardUpsertRequestTest.java

```diff
@@ -88,4 +95,36 @@ public void testStreaming() throws Exception {
         assertThat(request, equalTo(request2));
     }
 
+    @Test
+    public void testInvalidVersionNumber() throws IOException {
+        ShardId shardId = new ShardId("test", UUIDs.randomBase64UUID(), 1);
+        String[] assignmentColumns = new String[]{"id", "name"};
+        UUID jobId = UUID.randomUUID();
+        Reference[] missingAssignmentColumns = new Reference[]{ID_REF, NAME_REF};
+        ShardUpsertRequest request = new ShardUpsertRequest.Builder(
+            false,
+            false,
+            assignmentColumns,
+            missingAssignmentColumns,
+            jobId,
+            false
+        ).newRequest(shardId, "42");
+        request.validateConstraints(false);
+
+        request.add(123, new ShardUpsertRequest.Item(
+            "99",
+            null,
+            new Object[]{99, new BytesRef("Marvin")},
+            null));
+
+        expectedException.expect(IllegalStateException.class);
+        expectedException.expectMessage("Item with id '1' and version 9223372036854775807 has exceeded " +
+                                        "the maximum version number of 2147483647");
+        request.add(5, new ShardUpsertRequest.Item(
```

Comment:
afaik we could remove the usage of the request here completely as the check will be done at the item ctor or version setter. this would remove complexity of the test and makes it more obvious what is actually tested. also I'd prefer to test (also) the version setter as this is maybe the main case where an unsupported version is set.

Reply:
+1